src/sys/ioctl/mod.rs

```diff
@@ -138,17 +136,36 @@
 //! pub unsafe fn tcgets(fd: c_int, data: *mut termios) -> Result<c_int>;
 //! ```
 //!
-//! There is also a `bad none`, `bad write_int`/`bad write_ptr`, and `bad readwrite` variant that work
-//! similar to the standard `none`, `write_int`/`write_ptr`, and `readwrite` variants.
+//! The `ioctl_*_bad` variants should also be used when `ioctl` numbers are generated using the
+//! wrong macro. For example, the `KVM_CREATE_VM` ioctl is defined as follows in C:
```

Comment:
In this example, the ioctl is **not** using the wrong macro. The R/W in `_IOR`,  `_IOW`, `_IOWR` means that in addition to the parameter passed to ioctl, user memory is to be read/written. Since this ioctl doesn't pass a pointer, no additional user memory is involved, so there is nothing to be read or written.

Reply:
So the mapping should look like:
no argument (implicit 0) -> `_IO`
ulong argument -> `_IO`
pointer argument -> `_IOR`/`_IOW`/`_IOWR`

I believe that's what you're saying?

Reply:
Yes. This seems to also be the case for (some of the) BSDs as @myfreeweb pointed out.